src/mokka/src/integration-test/java/pl/hycom/mokka/emulator/mock/MockConfigurationControllerIT.java

```diff
@@ -0,0 +1,180 @@
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import org.springframework.web.context.WebApplicationContext;
+import pl.hycom.mokka.Application;
+
+import java.util.UUID;
+import java.util.stream.StreamSupport;
+
+import static io.restassured.RestAssured.expect;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.csrf;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+
+/**
+ * @author adam.misterski@hycom.pl
+ */
+@DirtiesContext
+@ActiveProfiles("test")
+@AutoConfigureMockMvc
+@AutoConfigureTestDatabase
+@SpringBootTest(classes = Application.class)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class MockConfigurationControllerIT {
+
+    @Autowired
+    protected WebApplicationContext wac;
+
+    @Autowired
+    private MockMvc mvc;
+
+    @Autowired
+    private MockConfigurationRepository mockConfigurationRepository;
+
+    @Autowired
+    private WireMockServer wm;
+
+    private static final String name = "test";
+    private static final String httpMethod = "GET";
+    private static final int status = 200;
+    private static final String path = "path";
+
+    private static final String CONTENT_TYPE_KEY = "Content-Type";
+
+    @BeforeEach
+    public void setUp() throws Exception{
+        wm.start();
+        RestAssured.port = wm.port();
+    }
+
+    @Test
+    @Order(1)
+    @WithMockUser(roles="ADMIN")
+    public void request_should_add_MockConfiguration_to_DB_and_WireMockMapping() throws Exception {
+
+        String requestBody = "{\"editMode\":true,\"isNew\":true,\"detailMode\":true,\"showDetails\":false,\"status\":\"" + status + "\",\"httpMethod\":\"" + httpMethod + "\",\"path\":\"" + path + "\",\"name\":\"" + name + "\",\"type\":\"string\",\"configurationContent\":{\"string\":{\"value\":\"test\"}}}\n";
+
+        mvc.perform(asyncRequest(put("/configuration")).contentType(MediaType.APPLICATION_JSON_VALUE).with(csrf())
+        .content(requestBody))
+        .andExpect(status().isOk());
+
+        Response response = RestAssured.get("/__admin/mappings");
+        expect().statusCode(200).header(CONTENT_TYPE_KEY, MediaType.APPLICATION_JSON_VALUE).response();
+        String mappings = response.getBody().jsonPath().get("mappings").toString();
+        UUID expectedUUID = UUID.nameUUIDFromBytes(mockConfigurationRepository.findAll().iterator().next().getId().toString().getBytes());
+
+        Assertions.assertTrue(mappings.contains(expectedUUID.toString()));
+        Assertions.assertEquals(1, StreamSupport.stream(mockConfigurationRepository.findAll().spliterator(), false).count());
```

Comment:
As I mentioned before (see `mockConfigurationRepository` field) we should abstract away from the implementation details.
So instead of using `mockConfigurationRepository` to "assert" I would propose the following:
* `put` on `/configuration" will return response with added MockConfiguration with assigned id.
* you may use it later: transform to uuid (this should not happen but lets leave it for now) and just call `/__admin/mappings/{uuid}`. In response you should receive StubMapping created from MockConfiguration. Just compare the most important parts : path, name, httpMethod